utils/utils.go

```diff
@@ -58,13 +61,51 @@ func AddLabel(obj metav1.Object, key string, value string) {
 	labels[key] = value
 }
 
-func AddAnnotation(obj metav1.Object, key string, value string) {
+func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
 		annotations = make(map[string]string)
 		obj.SetAnnotations(annotations)
+	} else if curr, found := annotations[key]; !found || curr != value {
+		annotations[key] = value
+		return true
 	}
-	annotations[key] = value
+	return false
+}
+
+func AddOwnerReference(owner, res client.Object, scheme *runtime.Scheme, isController bool) (bool, error) {
+	isChanged := false
+	gvk, err := apiutil.GVKForObject(owner, scheme)
+	if err != nil {
+		return isChanged, err
+	}
+	ref := metav1.OwnerReference{
+		APIVersion: gvk.GroupVersion().String(),
+		Kind:       gvk.Kind,
+		Name:       owner.GetName(),
+		UID:        owner.GetUID(),
+		Controller: &isController,
+	}
+	owners := res.GetOwnerReferences()
+	index := -1
+	for i := range owners {
+		if ref.APIVersion == owners[i].APIVersion &&
+			ref.Kind == owners[i].Kind &&
+			ref.Name == owners[i].Name &&
+			ref.UID == owners[i].UID {
+			index = i
+			break
+		}
+	}
+	if index != -1 {
+		owners[index] = ref
+		isChanged = false
```

Comment:
Why is this set to false? you are editing the annotation. 
This entire function is either appending an annotation or changing an existing one. 
Why do we want to track a change in the first place? I couldn't find a single call site that need this info and in fact all of the call sites are ignoring it using `_`

Reply:
We are using it over here
https://github.com/red-hat-storage/managed-fusion-agent/pull/20/files#diff-8755ab6909c6e17d04613ea6d890a985c19a2ac23db5893a220fb226918782d5R248